src/main/java/io/kestra/plugin/gcp/runner/GcpCloudRunTaskRunner.java

```diff
@@ -0,0 +1,409 @@
+                .setTemplate(ExecutionTemplate
+                    .newBuilder()
+                    .setTemplate(task)
+                    .setTaskCount(1)
+                    .build()
+                )
+                .putAllLabels(ScriptService.labels(runContext, "kestra-", true, true))
+                .build();
+
+            final String jobName = ScriptService.jobName(runContext);
+            final CreateJobRequest createJobRequest =
+                CreateJobRequest.newBuilder()
+                    // The job's parent is the region in which the job will run.
+                    .setParent(LocationName.of(renderedProjectId, region).toString())
+                    .setJob(job)
+                    .setJobId(jobName)
+                    .build();
+
+            Job jobCreated = jobsClient.createJobAsync(createJobRequest).get();
+            runContext.logger().info("Job created: {}", jobCreated.getName());
+
+            // Run the Job
+            Duration timeout = Optional.ofNullable(taskCommands.getTimeout()).orElse(this.waitUntilCompletion);
+            final JobName fullJobName = JobName.of(renderedProjectId, region, jobName);
+            final RunJobRequest runJobRequest =
+                RunJobRequest.newBuilder()
+                    .setName(fullJobName.toString())
+                    .setOverrides(RunJobRequest.Overrides
+                        .newBuilder()
+                        .setTaskCount(1)
+                        .setTimeout(com.google.protobuf.Duration
+                            .newBuilder()
+                            .setSeconds(timeout.getSeconds())
+                            .build()
+                        )
+                        .build()
+                    )
+                    .build();
+
+            OperationFuture<Execution, Execution> future = jobsClient.runJobAsync(runJobRequest);
+            Execution execution = future.getMetadata().get();
+            if (!isTerminated(execution)) {
+                runContext.logger().info("Waiting for execution completion: {}.", execution.getName());
+                execution = awaitJobExecutionTermination(executionsClient, execution.getName());
+            }
+            // Check for the job successful creation
+            if (isFailed(execution)) {
+                throw new TaskException(
+                    -1,
+                    taskCommands.getLogConsumer().getStdOutCount(),
+                    taskCommands.getLogConsumer().getStdErrCount()
+                );
+            }
+            String logFilter = String.format(
+                "logName=\"projects/%s/logs/run.googleapis.com\" labels.\"run.googleapis.com/execution_name\"=\"%s\"",
+                renderedProjectId,
+                execution.getName()
+            );
+
+            LogEntryServerStream stream = logging.tailLogEntries(Logging.TailOption.filter(logFilter));
+            try (LogTail ignored = new LogTail(stream, taskCommands.getLogConsumer())) {
```

Comment:
You should stream the log earlier, I would say before line 294 so before waiting for termination or logs will not arrives in real time.